### PR TITLE
Save gradient data as image

### LINKED ISSUE #199: Export grid distances and weights as a tiff file
## Use Case
We want to visualize distances and weights stored in the grid as a tiff file to explore 3D slices. Simularium example:
image

## Solution

1. Create array of values stored in the grid. This array will have the same size as the grid (may be high/low resolution depending on grid spacing)
2. Use the `ImageWriter` class to write these values to an ome tiff image.

### PR DESCRIPTION
**Pull request by mogres**
Note: Merge #215 before this PR

Problem
=======
We want to export grid distances and weights as 3D images.
Closes #199 

Solution
========
Used the `ImageWriter` class to export gradient distances and weights as 3D tiff images.

## Type of change
* New feature (non-breaking change which adds functionality)

Change summary:
---------------
* add option to save grid data as image 
* move voxelization code to environment
* add voxelization function for gradient distances and weights
* refactor image writer export code
* add recipe and config for testing

Steps to Verify:
----------------
1. `pack -r cellpack/tests/recipes/v2/test_nested_mesh_gradient.json -c cellpack/tests/packing-configs/test_gradient_data_image.json`
2. Open the output folder and navigate to `figures/gradient_data_figures` to view the exported image.

Screenshots:
-----------------------
Distances (mean intensity projection):
image

Weights (mean intensity projection):
image

Keyfiles (delete if not relevant):
-----------------------
1. `ImageWriter.py`
2. `Gradients.py`

### CODE REVIEW

**Comment by mogres** [comment]
Use `nbvv` or volume viewer to visualize the outputs of this data.

File: cellpack/tests/packing-configs/test_gradient_data_image.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "use_periodicity": false,
+    "show_sphere_trees": false,
+    "load_from_grid_file": true,
+    "save_converted_recipe": true,
+    "image_export_options": {
+        "hollow": false,
+        "voxel_size": [1,1,1],
+        "projection_axis": "z"
+    },
+    "save_gradient_data_as_image": true
```
Comment (meganrm) [inline cellpack/tests/packing-configs/test_gradient_data_image.json:23]: if this is a new config setting it should also have a default setting in the config loader. 
I also think, for clarity, any settings here that are defaults should be removed. The configs are ballooning a bit

File: cellpack/bin/pack.py [hunk 2]
```diff
@@ -70,6 +70,9 @@ def pack(recipe, config_path=None, analysis_config_path=None):
             recipe_version=recipe_data["version"],
             image_export_options=packing_config_data.get("image_export_options"),
             parallel=packing_config_data.get("parallel", False),
+            save_gradient_data_as_image=packing_config_data.get(
+                "save_gradient_data_as_image", False
+            ),
```
Comment (meganrm) [inline cellpack/bin/pack.py:75]: this is starting to make me think we should just sent the config into analysis instead of unpacking values here

**Review by meganrm — APPROVED** [review]
looks good to me, just one request that you add the new config setting to the config loader so we don't have mystery settings

**Review by rugeli — APPROVED** [review]
I ran the test recipe and was able to view the image without any problems. And I'm on board with adding `save_gradient_data_as_image` into config loader. Great work!

File: cellpack/tests/packing-configs/test_gradient_data_image.json (hunk 1 above)
Comment (mogres) [reply cellpack/tests/packing-configs/test_gradient_data_image.json]: Removed the default entries from the config file and added the new setting to `config_loader.py`

File: cellpack/bin/pack.py (hunk 2 above)
Comment (mogres) [reply cellpack/bin/pack.py:75]: Created an issue #227 to track this

**Pull request merged**